### DISCUSSION TITLE
How to configure Rspack to build app that uses preact/compat from esm.sh

### DISCUSSION THREAD

**Discussion by krutoo** [web-infra-dev/rspack#4279] [Q&A]
Hi, i want to build app that uses preact/compat from esm.sh in runtime (via script with type module)

Here is my attempt to build app without esm.sh but with preact/compat:
https://github.com/krutoo/preact-compat-rspack-esm-sh

How can a reconfigure this project for use preact from esm.sh?

**Comment by hardfist** [comment]
sorry, don't support http-import yet, you need write your own loader|plugin to handle this

**Reply by hardfist** [reply]
it's possible when we support https://webpack.js.org/configuration/experiments/#experimentsbuildhttp, but it's low priority now

**Reply by krutoo** [reply]
@hardfist Is this not possible even using `resolve.alias`?

i dont need to add modules from esm.sh to bundle

i just need to build app that imports things from remote source **in runtime**

**Reply by hardfist** [reply]
then external should work

**Reply by krutoo** [reply]
@hardfist I tried to use externals but I got errors in runtime

**Reply by hardfist** [reply]
give me a repro, and i will fix it for your

**Reply by krutoo** [reply]
@hardfist i updated main branch here: https://github.com/krutoo/preact-compat-rspack-esm-sh

**Reply by hardfist** [reply]
@krutoo  change the following configuration and it works, the related documentation is here https://webpack.js.org/configuration/externals/#:~:text=Let%27s%20load%20a%20lodash%20from%20CDN%3A
```js
  // this causes runtime error: Uncaught TypeError: preact__WEBPACK_IMPORTED_MODULE_0_.options is undefined
  externals: {
    'preact/debug': ["https://esm.sh/preact",'Preact']
  },
  externalsType: 'script'
```
